source/Symbol/Symtab.cpp

```diff
@@ -265,46 +265,48 @@ void Symtab::InitNameIndexes() {
         continue;
 
       const Mangled &mangled = symbol->GetMangled();
-      entry.cstring = mangled.GetMangledName().GetStringRef();
-      if (!entry.cstring.empty()) {
+      entry.cstring = mangled.GetMangledName();
+      if (entry.cstring) {
         m_name_to_index.Append(entry);
 
         // Now try and figure out the basename and figure out if the
         // basename is a method, function, etc and put that in the
         // appropriate table.
-        llvm::StringRef name = entry.cstring;
+        llvm::StringRef name = entry.cstring.GetStringRef();
         if (symbol->ContainsLinkerAnnotations()) {
           // If the symbol has linker annotations, also add the version without
           // the annotations.
           entry.cstring = ConstString(m_objfile->StripLinkerSymbolAnnotations(
-                                          entry.cstring))
-                              .GetStringRef();
+                                        entry.cstring.GetStringRef()));
           m_name_to_index.Append(entry);
         }
 
         const SymbolType symbol_type = symbol->GetType();
         if (symbol_type == eSymbolTypeCode ||
             symbol_type == eSymbolTypeResolver) {
-          if (name.size() >= 2 && name[0] == '_' && name[1] == 'Z' &&
-              ((name.size() < 3) || (name[2] != 'T' && // avoid virtual table, VTT structure, typeinfo
-                                 // structure, and typeinfo name
-               name[2] != 'G' && // avoid guard variables
-               name[2] != 'Z'))) // named local entities (if we eventually handle
-                                // eSymbolTypeData, we will want this back)
+          llvm::StringRef entry_ref(entry.cstring.GetStringRef());
```

Comment:
This is how the code is in llvm.org, and StripLinkerSymbolAnnotations strips off leading annotations that occur before the actual start of the mangled symbol. I suspect that was a bug fixed along the way, but since StripLinkerSymbolAnnotations only actually does anything on ELF, we it should be relatively safe to do this.